src/TPLinkDevice.php

```diff
@@ -62,15 +62,15 @@ public function sendCommand(array $command)
      */
     protected function connectToDevice()
     {
-        $this->client = stream_socket_client(
+        $this->client = @stream_socket_client(
             "tcp://" . $this->getConfig("ip") . ":" . $this->getConfig("port"),
             $errorNumber,
             $errorMessage,
-            5
+            $this->getConfig('timeout')
         );
 
         if ($this->client === false) {
-            throw new UnexpectedValueException("Failed to connect to {$this->deviceName}: $errorMessage ($errorNumber)");
+            throw new UnexpectedValueException("Failed connect to {$this->deviceName}: $errorMessage ($errorNumber)");
```

Comment:
I don't think removing the "to" helps here! Either "Connection failed to" or "Failed to connect to"